### ISSUE TITLE
[@typescript-eslint/no-use-before-define] 4.1.1 'React' was used before it was defined (.tsx files)

### ISSUE THREAD

**Issue by ievgennaida** [typescript-eslint/typescript-eslint#2579]
Just updated eslint plugin to the latest version and all React imports become invalid with an error:

`Line 7:8:  'React' was used before it was defined  @typescript-eslint/no-use-before-define`

Explanation from the related issue is not sufficient while @typescript-eslint/no-use-before-define is giving false positive warning.
Application is created with the react-create app scripts. 

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
    "@typescript-eslint/eslint-plugin": "4.1.1",
    "@typescript-eslint/parser": "4.1.1",
```
eslint v7.9.0

```TS
import { Button, Checkbox, TextField } from '@material-ui/core';
import React, { FunctionComponent, useEffect, useMemo } from 'react';
```

```JSON
{
  "env": {
    "browser": true,
    "es2020": true
  },
  "extends": [
    "eslint:recommended",
    "plugin:react/recommended",
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended"
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    },
    "ecmaVersion": 11
  },
  "plugins": ["react", "@typescript-eslint"],
  "rules": {
    "react/jsx-no-bind": [
      "error",
      {
        "allowArrowFunctions": true,
        "allowBind": false,
        "ignoreRefs": true
      }
    ],
    "@typescript-eslint/explicit-function-return-type": "off",
    "@typescript-eslint/no-unused-vars": "off",
    "no-redeclare": "off",
    "no-use-before-define": "off",
    "@typescript-eslint/no-use-before-define": ["error"],
    "react/jsx-filename-extension": [
      1,
      { "extensions": [".js", ".jsx", ".ts", ".tsx"] }
    ]
  },
  "settings": {
    "import/resolver": {
      "node": {
        "extensions": [".js", ".jsx", ".ts", ".tsx"]
      }
    },
    "react": {
      "version": "detect"
    }
  }
}
```

**Expected Result**

No warning

**Actual Result**

Warnings when running **npm run start**
```
./src/containers/Test.tsx
  Line 10:8:  'React' was used before it was defined  @typescript-eslint/no-use-before-define
```

**Comment by bradzacher** [comment]
Duplicate of the pinned issue
https://github.com/typescript-eslint/typescript-eslint/issues/2540
